app/controllers/application_controller.rb

```diff
@@ -1,10 +1,142 @@
 require './config/environment'
+require 'pry'
 
 class ApplicationController < Sinatra::Base
+  register Sinatra::ActiveRecordExtension
 
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "secret"
   end
 
+  get '/' do
+  	erb :index
+  end
+
+  get '/tweets' do
+  	if logged_in?
+  		erb :'/tweets/tweets'	
+ 	else
+	  	redirect '/login'
+  	end
+  end
+
+  get '/tweets/new' do
+  	if logged_in?
+	  	erb :'/tweets/create_tweet'
+	 else
+	  	redirect '/login'
+	 end
+  end
+
+  post '/tweets' do
+  	if params[:content] != ""
+	  	@tweet = Tweet.new(content: params[:content], user_id: session[:id])
+	  	@tweet.save
+	  	redirect "/tweets/#{@tweet.id}"
+	else
+		redirect "/tweets/new"
+	end
+  end
+
+  get '/tweets/:id' do
+  	if logged_in?
+	  	@tweet = Tweet.find_by_id(params[:id])
+	  	erb :'/tweets/show_tweet'
+	 else
+	  	redirect '/login'
+  	end
+  end
+
+  get '/tweets/:id/edit' do
+  	if logged_in?
+	  	@user = current_user
+	  	@tweet = Tweet.find_by_id(params[:id])
+	  	erb :'tweets/edit_tweet'
+	  else
+	  	redirect '/login'
+  	end
+  end
+
+  patch '/tweets/:id' do
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...